v3/newrelic/sqlparse/sqlparse.go

```diff
@@ -4,47 +4,168 @@
 package sqlparse
 
 import (
-	"regexp"
+	"bytes"
+	"fmt"
 	"strings"
+	"unicode"
 
 	newrelic "github.com/newrelic/go-agent/v3/newrelic"
 )
 
+var (
+	operations = map[string]string{
+		"select":   "from",
+		"delete":   "from",
+		"insert":   "into",
+		"update":   "",
+		"call":     "",
+		"create":   "",
+		"drop":     "",
+		"show":     "",
+		"set":      "",
+		"exec":     "",
+		"execute":  "",
+		"alter":    "",
+		"commit":   "",
+		"rollback": "",
+	}
+	updateModifiers = map[string]interface{}{
+		"low_priority": nil,
+		"ignore":       nil,
+		"or":           nil,
+		"rollback":     nil,
+		"abort":        nil,
+		"replace":      nil,
+		"fail":         nil,
+		"only":         nil,
+	}
+)
+
+// Extracts the table name from the given string.
 func extractTable(s string) string {
-	s = extractTableRegex.ReplaceAllString(s, "")
 	if idx := strings.Index(s, "."); idx > 0 {
 		s = s[idx+1:]
 	}
-	return s
+
+	var buffer bytes.Buffer
+
+	for _, c := range s {
+		if unicode.IsSpace(c) || strings.ContainsRune("`'(){}[]\"", c) {
+			continue
+		} else {
+			buffer.WriteRune(c)
+		}
+	}
+
+	return buffer.String()
 }
 
-var (
-	basicTable        = `[^)(\]\[\}\{\s,;]+`
-	enclosedTable     = `[\[\(\{]` + `\s*` + basicTable + `\s*` + `[\]\)\}]`
-	tablePattern      = `(` + `\s+` + basicTable + `|` + `\s*` + enclosedTable + `)`
-	extractTableRegex = regexp.MustCompile(`[\s` + "`" + `"'\(\)\{\}\[\]]*`)
-	updateRegex       = regexp.MustCompile(`(?is)^update(?:\s+(?:low_priority|ignore|or|rollback|abort|replace|fail|only))*` + tablePattern)
-	sqlOperations     = map[string]*regexp.Regexp{
-		"select":   regexp.MustCompile(`(?is)^.*?\sfrom` + tablePattern),
-		"delete":   regexp.MustCompile(`(?is)^.*?\sfrom` + tablePattern),
-		"insert":   regexp.MustCompile(`(?is)^.*?\sinto?` + tablePattern),
-		"update":   updateRegex,
-		"call":     nil,
-		"create":   nil,
-		"drop":     nil,
-		"show":     nil,
-		"set":      nil,
-		"exec":     nil,
-		"execute":  nil,
-		"alter":    nil,
-		"commit":   nil,
-		"rollback": nil,
+// Returns a string slice of query without trailing spaces.
+func skipSpace(query string) string {
+	for i, c := range query {
+		if !unicode.IsSpace(c) {
+			return query[i:]
+		}
 	}
-	firstWordRegex   = regexp.MustCompile(`^\w+`)
-	cCommentRegex    = regexp.MustCompile(`(?is)/\*.*?\*/`)
-	lineCommentRegex = regexp.MustCompile(`(?im)(?:--|#).*?$`)
-	sqlPrefixRegex   = regexp.MustCompile(`^[\s;]*`)
-)
+	return query
+}
+
+// Returns a string slice of query with trailing comments removed.
```

Comment:
This looks like it is skipping leading comments, not trailing comments.

Reply:
Fixed.